charm/src/charm.py

```diff
@@ -30,63 +35,55 @@ def __init__(self, *args: typing.Any) -> None:
-        super()._on_pebble_ready(_event)
-
-    def _copy_files(self) -> None:
-        """Copy files needed by git."""
-        if not self._container.can_connect():
-            return
-        if not self.config.get("git-repo") or not self.config.get("git-ssh-key"):
+        entries = []
+        known_notice = False
+        if event.notice.key.startswith("dns.local/write"):
+            fqdn = event.notice.last_data["fqdn"].strip("'").strip()
+            rdata = event.notice.last_data["rdata"].strip("'").strip()
+            entry = dns_record.RequirerEntry(
+                host_label=fqdn.split(".")[0],
+                domain=fqdn.split(".", 1)[1],
+                ttl=600,
+                record_class="IN",
+                record_type="TXT",
+                record_data=rdata,
+                uuid=uuid.uuid5(UUID_NAMESPACE, f"{fqdn} {rdata}"),
+            )
+            logger.debug("DNS record request: %s", entry)
+            entries = [entry]
+            known_notice = True
+
+        if event.notice.key.startswith("dns.local/remove"):
+            # For now, we remove everything
+            # since the default is no entries, we have nothing to do here.
+            known_notice = True
+
+        if not known_notice:
+            # We received an unknown notice, nothing to do with it.
+            logger.debug("Unknown notice: %s", event.notice.key)
             return
-        hostname = self.config.get("git-repo").split("@")[1].split("/")[0]
-        process = self._container.exec(["ssh-keyscan", "-t", "rsa", hostname])
-        output, _ = process.wait_output()
-        self._container.push(
-            KNOWN_HOSTS_PATH,
-            output,
-            encoding="utf-8",
-            make_dirs=True,
-            user=DJANGO_USER,
-            group=DJANGO_GROUP,
-            permissions=0o600,
-        )
-        self._container.push(
-            RSA_PATH,
-            self.config.get("git-ssh-key"),
-            encoding="utf-8",
-            make_dirs=True,
-            user=DJANGO_USER,
-            group=DJANGO_GROUP,
-            permissions=0o600,
-        )
 
-    def _on_collect_app_status(self, _: ops.CollectStatusEvent) -> None:
-        """Handle the status changes."""
-        if not self.config.get("git-repo"):
-            self.unit.status = ops.WaitingStatus("Config git-repo is required")
-            return
-        if not self.config.get("git-ssh-key"):
-            self.unit.status = ops.WaitingStatus("Config git-ssh-key is required")
+        dns_record_requirer_data = dns_record.DNSRecordRequirerData(dns_entries=entries)
+        if not self.model.unit.is_leader():
             return
+        try:
+            for relation in self.model.relations[self.dns_record.relation_name]:
+                self.dns_record.update_relation_data(relation, dns_record_requirer_data)
+        except ops.model.ModelError as e:
+            logger.error("ERROR while updating relation data: %s", e)
+            raise
```

Comment:
We should not be in active state if there's no relation and should checked before

Reply:
And the service should be put down when there's no relation